### PR TITLE
Update the README

### PR DESCRIPTION
**Pull request by tatsuya6502**
First of all, thank for working on this exiting project! I updated a few stuff in the README.

## Changes

* Fix the version of Rust nightly.
* Fix the file name of the pre-built llvm tarball for macOS arm64.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,21 +76,22 @@ is recommended to export XDG variables in general, but if you have not, just
 replace the usages of `$XDG_DATA_HOME` below with `$HOME/.local/share`, which is
 the usual default for this XDG variable.
 
-    mkdir -p $XDG_DATA_HOME/llvm/firefly
```
Comment (bitwalker) [inline README.md]: This change, and the introduction of the `mv` line below are incorrect - there is a difference here between Linux and macOS due to differing `tar` implementations, and on Linux we extract directly into the target directory. Can you revert the changes to this section? Everything else looks good

**Review by bitwalker — CHANGES_REQUESTED** [review]
Looks good overall, just the one change that needs to be reverted

File: README.md (hunk 1 above)
Comment (tatsuya6502) [reply README.md]: Thank you for spotting my mistakes. I reverted the changes in Linux commands.

**Review by bitwalker — APPROVED** [review]
